modules/gltf/gltf_document.cpp

```diff
@@ -5882,14 +5882,18 @@ void GLTFDocument::_generate_scene_node(Ref<GLTFState> p_state, const GLTFNodeIn
 	if (!gltf_node_name.is_empty()) {
 		current_node->set_name(gltf_node_name);
 	}
-	// Add the node we generated and set the owner to the scene root.
-	p_scene_parent->add_child(current_node, true);
-	if (current_node != p_scene_root) {
+	// Note: p_scene_parent and p_scene_root must either both be null or both be valid.
+	if (p_scene_root == nullptr) {
+		// If the root node argument is null, this is the root node.
+		p_scene_root = current_node;
+	} else {
+		// Add the node we generated and set the owner to the scene root.
+		p_scene_parent->add_child(current_node, true);
 		Array args;
 		args.append(p_scene_root);
 		current_node->propagate_call(StringName("set_owner"), args);
```

Comment:
An if statement got lost here.

It's important that the call to `set_owner` specifically still be guarded with a check that `(current_node != p_scene_root)`

Otherwise if you call `scene_root.set_owner(scene_root)` it will print an error or malfunction.

If you are 100% certain that `current_node` is never the same as `p_scene_root` in any of the 3 export modes, it would help if you wrote an explanation because it feels like it could happen especially in the single root case.

Reply:
It won't ever happen because `p_scene_root` will either be null or different from `current_node`. When generating the scene root node, we pass in `null` to this function.